v2/tools/generator/internal/config/type_transformer.go

```diff
@@ -55,37 +56,163 @@ type TypeTransformer struct {
+	if target.appliesCache == nil {
+		target.appliesCache = make(map[astmodel.Type]bool)
+	}
+
+	if result, ok := target.appliesCache[t]; ok {
+		return result
+	}
+
+	result := target.appliesToType(t)
+	target.appliesCache[t] = result
+	return result
+}
+
+func (target *TransformTarget) appliesToType(t astmodel.Type) bool {
+	if target == nil {
+		return true
+	}
+
+	inspect := t
+	if target.Optional {
+		// Need optional
+		opt, ok := astmodel.AsOptionalType(inspect)
+		if !ok {
+			// but don't have optional
+			return false
+		}
+
+		inspect = opt.Element()
+	}
+
+	if target.Name.IsRestrictive() {
+		if target.Group.IsRestrictive() || target.Version.IsRestrictive() {
+			// Expecting TypeName
+			if tn, ok := astmodel.AsTypeName(inspect); ok {
+				return target.appliesToTypeName(tn)
+			}
+
+			return false
+		}
+
+		// Expecting primitive type
+		if pt, ok := astmodel.AsPrimitiveType(inspect); ok {
+			return target.appliesToPrimitiveType(pt)
+		}
+
+		return false
+	}
+
+	if target.Map != nil {
+		// Expecting map type
+		if mp, ok := astmodel.AsMapType(inspect); ok {
+			return target.appliesToMapType(mp)
+		}
+
+		return false
+	}
+
+	return true
+}
+
+func (target *TransformTarget) appliesToTypeName(tn astmodel.TypeName) bool {
+	if !target.Name.Matches(tn.Name()) {
+		// No match on name
+		return false
+	}
+
+	g, v := tn.PackageReference.GroupVersion()
+
+	if target.Group.IsRestrictive() {
+		if !target.Group.Matches(g) {
+			// No match on group
+			return false
+		}
+	}
+
+	if target.Version.IsRestrictive() {
+
+		// Need to handle both full (v1beta20200101) and API (2020-01-01) formats
+		switch ref := tn.PackageReference.(type) {
+		case astmodel.LocalPackageReference:
+			if !ref.HasApiVersion(target.Version.String()) && !target.Version.Matches(v) {
+				return false
+			}
+		case astmodel.StoragePackageReference:
+			if !ref.Local().HasApiVersion(target.Version.String()) && target.Version.Matches(v) {
+				return false
+			}
+		default:
+			return false
+		}
+	}
+
+	return true
+}
+
+func (target *TransformTarget) appliesToPrimitiveType(pt *astmodel.PrimitiveType) bool {
+	if target.Name.Matches(pt.Name()) {
+		return true
+	}
+
+	// Special case, allowing config to use `any` as a synonym for `interface{}`
+	if strings.EqualFold(target.Name.String(), "any") &&
+		strings.EqualFold(pt.Name(), astmodel.AnyType.Name()) {
+		return true
+	}
+
+	return false
+}
+
+func (target *TransformTarget) appliesToMapType(mp *astmodel.MapType) bool {
+	return target.Map.Key.AppliesToType(mp.KeyType()) &&
+		target.Map.Value.AppliesToType(mp.ValueType())
+}
+
+func (target *TransformTarget) assignActualType(
+	descriptor string,
+	makeLocalPackageReferenceFunc func(group string, version string) astmodel.LocalPackageReference) error {
+	t, err := target.produceTargetType(descriptor, makeLocalPackageReferenceFunc)
+	if err != nil {
+		return err
+	}
+
+	target.actualType = t
+	return nil
+}
+
+func (target TransformTarget) produceTargetType(
```

Comment:
be consistent with ptr or non-ptr. This one is non-ptr while all the rest are ptr

Reply:
Fixed.